src/forwardBeamPipeBrazil.cpp

```diff
@@ -0,0 +1,570 @@
+        Material m_SS     = det.material("StainlessSteel");
+        Material m_vac    = det.material("Vacuum");
+        string   vis_name = x_det.visStr();
+
+        PlacedVolume pv_assembly;
+
+        double b0_hadron_tube_inner_r = 2.9 * dd4hep::cm;
+        double b0_hadron_tube_outer_r = 3.1 * dd4hep::cm;
+        double b0_hadron_tube_length  = 120.0 * dd4hep::cm;
+
+        double pipeThickness     = 5.0;
+
+        Double_t length[10];
+        Double_t innerXRadius[10];
+        Double_t innerYRadius[10];
+        Double_t outerXRadius[10];
+        Double_t outerYRadius[10];
+        Double_t rotationAngle[10];
+        Double_t xCenter[10];
+        Double_t yCenter[10];
+        Double_t zCenter[10];
+
+
+        double globRotationAngle = -0.0454486856; //This is the angle of the proton orbit from the end of B1APF to the beginning of B2PF
+
+        double b1APFEndPoint_z = 22062.3828; //location of proton orbit at b1APF exit -- in mm
+        double b1APFEndPoint_x = 654.3372; //location of proton orbit at b1APF exit -- in mm
+
+        double tmp_endpoint_z = 0.0;
+        double tmp_endpoint_x = 0.0;
+
+        //forumula -> Z = b1APFEndPoint_z+((0.5*elementLengt)*Cos(globRotationAngle))
+        //forumula -> X = b1APFEndPoint_z+((0.5*elementLengt)*Sin(globRotationAngle))
+
+        //------------------------------------------------------------------------------------
+        //primary pipe after B1APF, before neutral exit window + transition to smaller pipe
+        //rectangular cross-section!!!!!
+        //------------------------------------------------------------------------------------
+
+        length[0]            = 7615.486; //from VPC drawings, in mm
+        innerXRadius[0]      = 275.0;    //
+        innerYRadius[0]      = 175.0;    //
+        outerXRadius[0]      = innerXRadius[0] + pipeThickness; //
+        outerYRadius[0]      = innerYRadius[0] + pipeThickness; //
+
+        xCenter[0]           = -1*(b1APFEndPoint_x+((0.5*length[0])*TMath::Sin(-globRotationAngle))); //-745.20328;
+        xCenter[0]           = xCenter[0]; //shift of center location to account for offset
+        yCenter[0]           = 0.0;
+        zCenter[0]           = (b1APFEndPoint_z+((0.5*length[0])*TMath::Cos(-globRotationAngle)));//24060.3176;
+        rotationAngle[0]     = globRotationAngle;
+
+        tmp_endpoint_z = zCenter[0]+((0.5*length[0])*TMath::Cos(-globRotationAngle));
+        tmp_endpoint_x = -1*xCenter[0]+((0.5*length[0])*TMath::Sin(-globRotationAngle));
+
+        double windowRadius = 110.0;
+
+
+        //------------------------------------------------------------------------------------
+        //first small pipe section, between primary vessel and RP station 1
+        //rectangular cross-section!!!!!
+        //------------------------------------------------------------------------------------
+
+        length[1]            = 2780.273; // from VPC drawings
```

Comment:
Can you add more details to the reference here: VPC drawings which version, date, location where this can be found?

Reply:
Done.